src/main/java/org/humancellatlas/ingest/patch/Patch.java

```diff
@@ -0,0 +1,27 @@
+package org.humancellatlas.ingest.patch;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.humancellatlas.ingest.core.AbstractEntity;
+import org.humancellatlas.ingest.core.MetadataDocument;
+import org.humancellatlas.ingest.submission.SubmissionEnvelope;
+import org.springframework.data.mongodb.core.mapping.DBRef;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.rest.core.annotation.RestResource;
+
+import java.util.Map;
+
+@Data
+@AllArgsConstructor
+@Document
+public class Patch<T extends MetadataDocument> extends AbstractEntity {
```

Comment:
is it not enough to just store the jsonPatch and the update document here?

Reply:
We could but storing references to the envelope allows for a fast query to findAllPatchesBySubmisssion() if we need it, as well as findAllByOriginalDocument(). Keeping track of them adds no extra complexity since they're already in scope when generating a Patch